quipucords/api/serializers.py

```diff
@@ -26,3 +28,23 @@ class HostCredentialSerializer(ModelSerializer):
     class Meta:
         model = HostCredential
         fields = '__all__'
+
+    def validate(self, attrs):
```

Comment:
Can we make the validate method simpler? I think something like

`ssh_keyfile = 'ssh_keyfile' in attrs and attrs['ssh_keyfile']`, and similar for `password`. Then `ssh_keyfile` would be either `False`, if it wasn't in attrs, `None`, if it was in attrs but None, or a nonempty string with a path. The conditionals could use implicit conversion to bool to check the False and None cases at the same time.